From a2d0c6c0d380692afdd5eae4d658e6f1801dc709 Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Fri, 20 Jul 2018 12:05:46 +0200
Subject: [PATCH] swconfig: speed up ATU flush for MV88E6321

ATU flush functionality is needed by PROFINET.
The requirement is to start ATU flush and return
back to application (PROFINET stack) as quick as passible.
It is not necessary to wait until switch is done with flushing.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 drivers/net/phy/mv88e6321-cfg.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/net/phy/mv88e6321-cfg.c b/drivers/net/phy/mv88e6321-cfg.c
index d35617f..35bc56a 100644
--- a/drivers/net/phy/mv88e6321-cfg.c
+++ b/drivers/net/phy/mv88e6321-cfg.c
@@ -934,7 +934,7 @@ static inline int mv88e6321_wait_atu_ready(struct mii_bus *bus)
 			goto out;
 		}
 
-		usleep_range(1000, 2000);
+		usleep_range(500, 1000);
 	}
 
 	ret = -ETIMEDOUT;
@@ -980,10 +980,6 @@ static int mv88e6321_atu_flush(struct mii_bus *bus)
 		goto out;
 	}
 
-	ret = mv88e6321_wait_atu_ready(bus);
-	if (ret)
-		goto out;
-
 out:
 	return ret;
 }
-- 
2.7.4

